src/azure-cli-core/azure/cli/core/commands/command_operation.py

```diff
@@ -0,0 +1,489 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import argparse
+from six import string_types, get_method_function
+from azure.cli.core import AzCommandsLoader, EXCLUDED_PARAMS
+from knack.introspection import extract_args_from_signature, extract_full_summary_from_signature
+from knack.arguments import CLICommandArgument, ignore_type
+
+
+class BaseCommandOperation:
```

Comment:
The word `Operation` seems redundant to me and makes the functionality difficult to understand. `BaseCommand` should suffice in my opinion.